test/e2e/scheduling/predicates.go

```diff
@@ -900,16 +892,38 @@ func getRequestedCPU(pod v1.Pod) int64 {
 	return result
 }
 
-func waitForScheduler() {
-	// Wait a bit to allow scheduler to do its thing
-	// TODO: this is brittle; there's no guarantee the scheduler will have run in 10 seconds.
-	framework.Logf("Sleeping 10 seconds and crossing our fingers that scheduler will run in that time.")
-	time.Sleep(10 * time.Second)
+// removeTaintFromNodeAction returns a closure that removes the given taint
+// from the given node upon invocation.
+func removeTaintFromNodeAction(cs clientset.Interface, nodeName string, testTaint v1.Taint) func() error {
+	return func() error {
+		framework.RemoveTaintOffNode(cs, nodeName, testTaint)
```

Comment:
why does this return error, then it's always nil... Looks like there is a Failf buried there.